vc_vidyo/indico_vc_vidyo/api/client.py

```diff
@@ -77,11 +77,28 @@ def create_room_object(self, **kwargs):
 
     @raises_api_error
     def find_room(self, extension):
+        from indico_vc_vidyo.plugin import VidyoPlugin
         filter_ = self.client.factory.create('Filter')
         filter_.query = extension
+        filter_.limit = 40
         filter_.dir = 'DESC'
 
-        return self.soap.getRooms(filter_).room
+        target_room = None
+        counter = 0
+
+        while True:
+            filter_.start = counter*filter_.limit
+            rooms = self.soap.getRooms(filter_).room
+            for room in rooms:
+                if int(room.extension) == int(extension):
+                    VidyoPlugin.logger.debug('Room: %s has been found.', room)
+                    target_room = room
```

Comment:
Why not returning `room` from here directly? That way you can get rid of `break` and `if target_room` down in `L99`.